roles/rhsm/tasks/1_cleanrepos.yml

```diff
@@ -0,0 +1,12 @@
+- name: Clean Repos | Get a list of all the files in /etc/yum.repos.d
+  find:
+    paths: /etc/yum.repos.d
+    patterns: "*.repo"
+  register: repo_files
+
+- name: Clean Repos | Remove files in /etc/yum.repos.d not redhat.repo
+  file:
+    path: "{{ item.path }}"
+    state: absent
+  when: item.path != "/etc/yum.repos.d/redhat.repo"
```

Comment:
this is not idempotent for disconnected setups

Reply:
added a check for that file @evgeni